### PR TITLE
Route viewer visual tweaks

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR makes improves the route viewer layout, so route numbers are not shrunk on narrow screens and long route numbers are properly clipped.

### CODE REVIEW

**Review by amy-corson-ibigroup — APPROVED** [review]
This works really well! Just had a comment regarding margins and spacing, but it's solving the clipping problem beautifully on smaller screens.

File: lib/components/viewers/RouteRow.js [hunk 1]
```diff
@@ -24,41 +24,34 @@ export const RouteRowButton = styled(Button)`
...
-const RouteNameElement = styled(Label)`
-  background-color: transparent;
-  color: inherit;
-  flex: 0 1 auto;
+const RouteNameElement = styled.span`
+  flex-shrink: 0;
   font-size: 16px;
   font-weight: 400;
   margin-left: 8px;
-  margin-top: 2px;
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply lib/components/viewers/RouteRow.js:51]: I like to have the route number close to the mode icon, but increasing the margin between the route number and route long name (e.g. to 16px or 20px) makes sense.

File: lib/components/narrative/metro/default-route-renderer.tsx [hunk 2]
```diff
@@ -2,16 +2,21 @@ import { Leg } from '@opentripplanner/types'
...
-  text-align: center;
+  border-top: 5px solid #${(props) => props.color};
   display: inline-block;
-  max-width: 12vw;
+  margin-top: -2px;
+  padding: 3px 7px;
+
+  /* Below is for route names that are too long: cut-off and show ellipsis. */
+  max-width: 150px;
+  overflow: hidden;
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/metro/default-route-renderer.tsx:15]: Not sure if it's this line, but there is some trouble in the itinerary body that wasn't there previously.
Screenshot 2022-12-06 at 9 58 50 AM

**Review by miles-grant-ibigroup — APPROVED** [review]
Looking good -- one minor bug but nothing blocking. Thanks for the cleanup!

File: lib/components/narrative/metro/default-route-renderer.tsx (hunk 2 above)
Comment (binh-dam-ibigroup) [reply lib/components/narrative/metro/default-route-renderer.tsx:15]: Is this on Safari? This also appears in the dev branch.

Comment (miles-grant-ibigroup) [reply lib/components/narrative/metro/default-route-renderer.tsx:15]: Might've been a cache issue when I switched back to dev to check. If it's not a regression then we can leave as is

Comment (binh-dam-ibigroup) [reply lib/components/narrative/metro/default-route-renderer.tsx:15]: That probably comes from this line: https://github.com/opentripplanner/otp-react-redux/blob/20ca3fa4531a28a9a7d0c0fb2101f0db9a1c03a4/lib/components/narrative/metro/route-block.tsx#L30

**Pull request merged**